### ISSUE TITLE
Issues with Premiumize

### ISSUE THREAD

**Issue by samh06** [Gabisonfire/knightcrawler#93]
**Describe the bug**
Playing media with stremio through premiumize is mostly broken:
![image](https://github.com/Gabisonfire/knightcrawler/assets/54530346/e7135d2e-7bae-4ebb-b514-52e77d2c1d5d)

For probably 95% of media this error screen appears.

In the logs it either will say:
Failed Premiumize adding torrent "No recent torrent found"
or
Failed Premiumize adding torrent {}

**To Reproduce**
Install Knightcrawler stremio addon with a premiumize account linked.
Play any media.
Almost certain chance it will not work.

**Expected behavior**
To play the media or say the media is being downloaded.

**Logs**
[logs on pastebin](https://pastebin.com/QyQL7YLG)

**Hardware:**
 - Running from a hosted Knightcrawler instance on Elfhosted.

**Additional context**
The logs aren't detailed because it isn't particularly clear anywhere to how to change the log level.
The files are actually being added to my premiumize instance, and can be played from the 'View Content' button on the banner that shows up after adding an addon. 
This isn't an issue when played through the original torrentio.
Out of probably 40 to 60 files i tried playing I only had 3 actually work, and couldn't really see any pattern with them.
I had this happen on both the public and my hosted instance of knightcrawler.

**Comment by prom3theu5** [comment]
@Gabisonfire We probably need to bring in upstream changes for the premiumize moch here - I'm not sure, I don't use premiumize so cant test.

**Comment by Gabisonfire** [comment]
@prom3theu5 Good idea, I'm just wondering if this was here from day one. I don't have an acocunt either.

**Comment by iPromKnight** [comment]
I don't have a premiumize account, so am unable to tests sorry - but I think its just a case of bringing in the upstream moch like Gabi has done in #97

**Comment by purple-emily** [comment]
@asylumexp can you test this fix? Are you able to pull and build the branch and try it?

**Comment by asylumexp** [comment]
> @asylumexp can you test this fix? Are you able to pull and build the branch and try it?

I'll have a try of it

**Comment by asylumexp** [comment]
> > @asylumexp can you test this fix? Are you able to pull and build the branch and try it?
> 
> I'll have a try of it

ive been trying for a bit but i keep getting an error saying my premiumize key is incorrect
![image](https://github.com/Gabisonfire/knightcrawler/assets/54530346/8e71e20e-8ef9-4dfb-89eb-a836a379dc4a)
ive reinstalled the addon several times and its definitely the right API key

**Comment by purple-emily** [comment]
@asylumexp this fix won’t be live on ElfHosted. It would need to be locally built.

@Gabisonfire you want to merge the PR and see?

**Comment by asylumexp** [comment]
> @asylumexp this fix won’t be live on ElfHosted. It would need to be locally built.
> 
> @Gabisonfire you want to merge the PR and see?

I talked to @funkypenguin and apparently should already have it according to him.
![image](https://github.com/Gabisonfire/knightcrawler/assets/54530346/61949a43-93bf-433d-8da1-302834d2c23a)
Perhaps im misunderstanding something but I thought that would mean the patch was on my instance

**Comment by Gabisonfire** [comment]
He pushed the update about 2 hours ago,

**Comment by funkypenguin** [comment]
Usually ElfHosted apps get updated during a daily release/maintenance cycle, but alpha-channel users get updates as they're available. So yes, @asylumexp should have the latest code in his hosted instance, and it's also on https://torrentio.elfhosted.com / https://knightcrawler.elfhosted.com

**Comment by Gabisonfire** [comment]
@asylumexp It this still an issue?
